maven-core/src/main/java/org/apache/maven/internal/transformation/ConsumerPomArtifactTransformer.java

```diff
@@ -165,31 +178,50 @@ private ConsumerPomArtifact(MavenProject mavenProject, Path target, RepositorySy
                     target,
                     transformer(session));
         }
+    }
 
-        private static BiConsumer<Path, Path> transformer(RepositorySystemSession session) {
-            TransformerContext context = (TransformerContext) session.getData().get(TransformerContext.KEY);
-            return (src, dest) -> {
-                try (InputStream inputStream = transform(src, context)) {
-                    Files.createDirectories(dest.getParent());
-                    Files.copy(inputStream, dest, StandardCopyOption.REPLACE_EXISTING);
-                } catch (XMLStreamException | IOException e) {
-                    throw new RuntimeException(e);
-                }
-            };
-        }
+    BiConsumer<Path, Path> transformer(RepositorySystemSession session) {
+        TransformerContext context = (TransformerContext) session.getData().get(TransformerContext.KEY);
+        return (src, dest) -> {
+            try {
+                Files.createDirectories(dest.getParent());
+                transform(src, dest, context);
+            } catch (XMLStreamException | IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Unless it can be proved these exceptions cannot be thrown, converting to a runtime exception is not OK. If this means not using a lambda here, that's fine.

Reply:
So you want me to define an interface with `throw Exception` just to avoid using unchecked exception ?
Worse, define a new interface + exception type, both would only be used in this context, just to wrap another checked exception for the sole purpose of not using unchecked exceptions ?
Again, that's a more general discussion we can have, but I don't think it's the right place.  Feel free to start a discussion thread on exception handling in the v4 api on the mailing list.

Reply:
This discussion does not belong to this PR, as this is a design decision in the v4 api and can't be changed in this smaller PR.